### ISSUE TITLE
Rename "task" to "job" and "pipeline" to "task"

### ISSUE THREAD

**Issue by JeanMertz** [blendle/automaat#28]
After using and talking about Automaat for a while now, I think the naming scheme is confusing – especially for non-technical people – and thus should be changed.

Specifically, right now a _pipeline_ is a collection of steps that you can run, and when you run a pipeline, it creates a _task_ based on the configuration of that pipeline.

Technically, the name _pipeline_ makes sense, since you are chaining together a set of steps which each run a single processor with some configuration attached. But, when talking about it within the UI, in the abstract, without (having to) knowing how a pipeline works, this analogy breaks down.

Instead, talking about a _task_ makes things much more clear, as the dictionary defines a task as _"a piece of work to be done or undertaken."_, which fits perfectly in this context.

Similarly, whereas tasks currently mean "a pipeline that is actively running", it makes more sense to call this a _job_: _"the object or material on which work is being done "_.

**So, in conclusion:** the term **pipeline** will disappear, the term **task** will take the role of a pipeline, and the term **job** will take the role of a task.

This work was already done for the web client in 2d9bc81bec96ae606c2680090469376f3926b68a, so this issue is to track the change of the terms in the rest of the code base (mostly the API server).
